crates/sui-json-rpc-api/src/rpc_types.rs

```diff
@@ -1371,3 +1444,57 @@ pub struct MoveCallParams {
     pub type_arguments: Vec<SuiTypeTag>,
     pub arguments: Vec<SuiJsonValue>,
 }
+
+#[derive(Serialize, Deserialize, JsonSchema, Debug)]
+#[serde(rename = "EventFilter")]
+pub enum SuiEventFilter {
+    Package(ObjectID),
+    Module(String),
+    /// Move StructTag string value of the event type e.g. `0x2::devnet_nft::MintNFTEvent`
+    MoveEventType(String),
```

Comment:
Isn't this redundant?  `MoveEventType` == `All(vec![Package(..), Module(...)])`

Reply:
this will be redundant when the only allow event from the same module restriction is enforced in Move code, we can remove the usage of Package and Module filter for MoveEvents.